src/components/Database/DataViewer/common/fetch.ts

```diff
@@ -90,7 +90,7 @@ function realtimeToViewModel(
   queryParams: QueryParams
 ) {
   const query = applyQuery(ref, queryParams);
-  return once(ref).pipe(
+  return once(query).pipe(
```

Comment:
Code LGTM but I don't get the reason behind this section, especially the switchMap below, and I'd like to hear some thoughts from @tstirrat .

Reply:
Ok will talk offline about it.

Reply:
I actually think you should remove this line change. This is an initial detection to work out if a filter can be applied at the node.. so it's better to `once(ref)`.. if the once(query) returned no nodes, it would then give strange results.

Reply:
If you filtered to zero children, then it would yield `snap.hasChildren() === false`, and then it would return `toObservable(ref)` with the ref (not query).. so filters that yield no results will **inactivate themselves**.